controller/src/main/java/org/jboss/as/controller/operations/global/GlobalOperationHandlers.java

```diff
@@ -296,28 +306,178 @@ private void safeExecute(final PathAddress address, final PathAddress base, fina
          *                                  tree is optional if it is associated with a wildcard or multi-segment
          *                                  element in {@code address}
          */
-        private void execute(final PathAddress address, final PathAddress base, final OperationContext context, boolean ignoreMissing) {
+        private void execute(final PathAddress base, final PathAddress remaining, final OperationContext context,
+                             final ImmutableManagementResourceRegistration registration, final boolean ignoreMissing) {
+
+            // Check whether the operation needs to be dispatched to a remote proxy
+            if (registration.isRemote()) {
+                // make sure the target address does not contain the unresolved elements of the address
+                final ModelNode remoteOp = operation.clone();
+                final PathAddress fullAddress = base.append(remaining);
+                remoteOp.get(OP_ADDR).set(fullAddress.toModelNode());
+                // Temp remote result
+                final ModelNode resultItem = new ModelNode();
+
+                final OperationStepHandler delegate = registration.getOperationHandler(PathAddress.EMPTY_ADDRESS, operation.require(OP).asString());
+                context.addStep(resultItem, remoteOp, new OperationStepHandler() {
+                    @Override
+                    public void execute(OperationContext context, ModelNode operation) throws OperationFailedException {
+                        try {
+                            // Execute the proxy step handler in a separate step
+                            // so we have the final response available to our ResultHandler
+                            ControllerLogger.MGMT_OP_LOGGER.tracef("sending ModelAddressResolver request %s to remote process using %s",
+                                    operation, delegate);
+
+                            final AtomicBoolean filtered = new AtomicBoolean(false);
+
+                            context.addStep(new OperationStepHandler() {
+                                @Override
+                                public void execute(OperationContext context, ModelNode operation) throws OperationFailedException {
+                                    try {
+                                        delegate.execute(context, operation);
+                                    } catch (UnauthorizedException e) {
+                                        // equivalent to the resource not existing
+                                        // Just report the failure to the filter and complete normally
+                                        filteredData.addReadRestrictedResource(base);
+                                        filtered.set(true);
+                                        ControllerLogger.MGMT_OP_LOGGER.tracef("Caught UnauthorizedException in %s from %s", this, delegate);
```

Comment:
`toString()` probably should be overridden here otherwise you'll end up with something like `org.jboss.as.controller.operations.global.GlobalOperationHandlers.ModelAddressResolver.$1@xxxxx`. Maybe not super important I guess. I'll keep reading further.